apps/frontend/src/libs/components/balance-wheel-chart/libs/helpers/draw-dots/draw-dots.helper.ts

```diff
@@ -0,0 +1,63 @@
+import { HALF_PI, TAU } from "~/libs/constants/constants.js";
+import {
+	type ChartArea,
+	type ChartMeta,
+	type RadialLinearScale,
+} from "~/libs/types/types.js";
+
+import { DOT_RADIUS, START_ANGLE } from "../../constants/constants.js";
+import { ChartGraphicsColor, WheelCenterDistance } from "../../enums/enums.js";
+import { type PolarAreaType } from "../../types/types.js";
+import { getDotOffsetX } from "../get-dot-offset-x/get-dot-offset-x.helper.js";
+import { getDotOffsetY } from "../get-dot-offset-y/get-dot-offset-y.helper.js";
+import { getGraphicsOffset } from "../get-graphics-offset/get-graphics-offset.js";
+
+const drawDots = ({
+	chartArea,
+	context,
+	index,
+	meta,
+	middleAngle,
+	scale,
+}: {
+	chartArea: ChartArea;
+	context: CanvasRenderingContext2D;
+	index: number;
+	meta: ChartMeta<PolarAreaType>;
+	middleAngle: number;
+	scale: RadialLinearScale;
+}): void => {
```

Comment:
Let's move it to a separate type and place above the drawDots function. Let's name it ChartPositionData or you may pick your own name